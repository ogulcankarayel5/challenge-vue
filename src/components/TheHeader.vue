<template>
  <header class="header">
    <p class="rightText">hepsiburada.com</p>
    <p class="leftText">LinkVote Challenge</p>
  </header>
</template>

<script>
export default {
  name: "Header",
};
</script>

<style lang="sass" scoped>
.header

    font-size: 2rem
    margin-left: auto
    margin-right: auto
    width: 100%
    max-width: 1280px
    border-bottom: 1px solid #000
    display: flex
    justify-content: space-between

    & .rightText
        color: #FF6000
</style>
