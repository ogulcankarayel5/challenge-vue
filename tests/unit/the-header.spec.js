import { mount, shallowMount } from "@vue/test-utils";
import TheHeader from "@/components/TheHeader.vue";

describe("TheHeader.vue", () => {
  it("snapshot", () => {
    const wrapper = shallowMount(TheHeader);

    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders component correctly", () => {
    const wrapper = mount(TheHeader);
    expect(wrapper.find(".rightText").text()).toEqual("hepsiburada.com");
    expect(wrapper.find(".leftText").text()).toEqual("LinkVote Challenge");
  });
});
