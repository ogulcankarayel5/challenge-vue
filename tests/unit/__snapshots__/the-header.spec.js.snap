// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader.vue snapshot 1`] = `
<header
  class="header"
>
  <p
    class="rightText"
  >
    hepsiburada.com
  </p>
  <p
    class="leftText"
  >
    LinkVote Challenge
  </p>
</header>
`;
